components/org.wso2.carbon.identity.scim.common/src/main/java/org/wso2/carbon/identity/scim/common/listener/SCIMTenantCreationEventListener.java

```diff
@@ -0,0 +1,132 @@
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.charon.core.schema.SCIMConstants;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+
+/**
+ * This is an implementation of TenantMgtListener. This is used
+ * to generate SCIM attributes for tenant admin users.
+ */
+public class SCIMTenantCreationEventListener implements TenantMgtListener {
+
+    private static Log log = LogFactory.getLog(SCIMTenantCreationEventListener.class);
+
+    @Override
+    public void onTenantCreate(TenantInfoBean tenantInfoBean) throws StratosException {
+
+    }
+
+    @Override
+    public void onTenantUpdate(TenantInfoBean tenantInfoBean) throws StratosException {
+
+    }
+
+    @Override
+    public void onTenantDelete(int i) {
+
+    }
+
+    @Override
+    public void onTenantRename(int i, String s, String s1) throws StratosException {
+
+    }
+
+    @Override
+    public void onTenantInitialActivation(int i) throws StratosException {
+
+        setTenantAdminSCIMAttributes(i);
+    }
+
+    @Override
+    public void onTenantActivation(int i) throws StratosException {
+
+    }
+
+    @Override
+    public void onTenantDeactivation(int i) throws StratosException {
+
+    }
+
+    @Override
+    public void onSubscriptionPlanChange(int i, String s, String s1) throws StratosException {
+
+    }
+
+    @Override
+    public int getListenerOrder() {
+
+        return 0;
+    }
+
+    @Override
+    public void onPreDelete(int i) throws StratosException {
+
+    }
+
+    /**
+     * Set tenant admins SCIM attributes on tenant initial activation.
+     *
+     * @param tenantId a tenantId
+     */
+    private void setTenantAdminSCIMAttributes(int tenantId) {
+
+        try {
+            UserStoreManager userStoreManager = (UserStoreManager) SCIMCommonComponentHolder.getRealmService()
+                    .getTenantUserRealm(tenantId).getUserStoreManager();
+
+            if (userStoreManager.isSCIMEnabled()) {
+                Map<String, String> claimsMap = new HashMap<String, String>();
+                String adminUsername = ClaimsMgtUtil.getAdminUserNameFromTenantId(IdentityTenantUtil.getRealmService(),
+                        tenantId);
+                String id = UUID.randomUUID().toString();
+                Date date = new Date();
+                String createdDate = SCIMCommonUtils.formatDateTime(date);
+
+                claimsMap.put(SCIMConstants.META_CREATED_URI, createdDate);
+                claimsMap.put(SCIMConstants.USER_NAME_URI, adminUsername);
+                claimsMap.put(SCIMConstants.META_LAST_MODIFIED_URI, createdDate);
+                claimsMap.put(SCIMConstants.ID_URI, id);
+
+                userStoreManager.setUserClaimValues(adminUsername, claimsMap,
+                        UserCoreConstants.DEFAULT_PROFILE);
+            }
+        } catch (Exception e) {
+            String msg = "Error in adding claims to the tenant admin. tenant ID: " + tenantId;
```

Comment:
Change the msg as Error while adding SCIM metadata to the tenant admin in tenant ID :